src/main/java/bowling/domain/FinalFrame.java

```diff
@@ -0,0 +1,80 @@
+package bowling.domain;
+
+import java.util.List;
+import java.util.Optional;
+
+public class FinalFrame implements Frame {
+    private static final int MAX_PIN_COUNT_SIZE = 3;
+    private static final int MIN_PIN_COUNT_FOR_THIRD = 10;
+
+    private PinCounts pinCounts;
+
+    public FinalFrame() {
+        pinCounts = new PinCounts();
+    }
+
+    @Override public boolean addPinCount(int pinCount) {
+        if (!isAddable(pinCount)) {
+            return false;
+        }
+
+        return pinCounts.add(pinCount);
+    }
+
+    private boolean isAddable(int pinCount) {
+        if (isDone()) {
+            return false;
+        }
+
+        if (hasThirdChance()) {
+            return true;
+        }
+
+        Optional<PinCount> lastPinCount = pinCounts.getLast();
+        return lastPinCount.map(pc -> !pc.isOverMaxAfterAdd(pinCount))
+                .orElse(true);
+    }
+
+    private boolean hasThirdChance() {
+        PinCount firstPinCount = pinCounts.getFirst()
+                .orElse(PinCount.empty());
+        PinCount secondPinCount = pinCounts.getSecond()
+                .orElse(PinCount.empty());
+
+        if (firstPinCount.isStrike()) {
+            return true;
+        } else return secondPinCount.isSpare();
+    }
+
+    private boolean isPinCountsFull() {
+        return pinCounts.size() == MAX_PIN_COUNT_SIZE;
+    }
+
+    @Override public int getScore() {
+        return pinCounts.getPinCountTotal();
+    }
+
+    @Override public boolean isDone() {
+        if (isPinCountsFull()) {
```

Comment:
isPinCountsFull을 PinCounts로 옮긴다면 값을 조회하여 full 여부를 판단하는게 아니라 PinCounts에 직접 물어보는 형태가 가능할 것 같네요 :)
```suggestion
        if (pinCounts.isFull()) {
```